### PR TITLE
Survey methods

### PR DESCRIPTION
**Pull request by LouisThedroux**
## Links to Jira Tickets

NA 

## Description of Changes
New migration file that includes a new list of survey methods.
Macgregor and I had a meeting with the wider team about existing SPI survey methods and what ones should go into SIMS. We refined the list and I edited some/removed some etc to tidy up the list and match the structure and format of SIMS.

### CODE REVIEW

File: database/src/migrations/20250527114701_update_survey_methods.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    WHERE name IN ('Aerial transect', 'Ground transect', 'Aquatic transect', 'Underwater transect');  
+
+  ----------------------------------------------------------------------------------------
+  --Adding more values to the method lookup table. 
+  ----------------------------------------------------------------------------------------
+    INSERT into method_lookup (name, description)
+    VALUES
+    (
+        'Gill Net',
+        'a fishing net that hangs vertically in the water, trapping fish by their gills when they try to swim through its mesh openings.'
```
Comment (mauberti-bc) [inline database/src/migrations/20250527114701_update_survey_methods.ts]: Capitalize the first letter

File: database/src/migrations/20250527114701_update_survey_methods.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+
+  SET SEARCH_PATH=biohub;
+
+  ----------------------------------------------------------------------------------------
+  --Deleting previous rows that no longer apply to the method_lookup table
+  ----------------------------------------------------------------------------------------
+    DELETE FROM method_lookup
```
Comment (mauberti-bc) [inline database/src/migrations/20250527114701_update_survey_methods.ts]: We should probably change any existing Aerial transects, ground transects, aquatic transects, and underwater transects to "Visual observation". If these transect methods exist in the survey_sample_method table, trying to drop them from the method_lookup table will fail with a foreign key constraint error.

Try adding this to the end of this migration, after having created the Visual encounter method if it doesn't already exist, followed by the deletion of the transect methods part. (Haven't tested the below sql but looks fine)

```
UPDATE survey_sample_method
SET method_lookup_id = (SELECT method_lookup_id FROM method_lookup WHERE name = 'Visual encounter')
WHERE method_lookup_id IN (SELECT method_lookup_id FROM method_lookup WHERE name IN ('Aerial transect', 'Ground transect', 'Aquatic transect', 'Underwater transect'));
```

File: database/src/migrations/20250527114701_update_survey_methods.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        'Fish Wheel',
+        'A fish wheel is used in sample collection to automatically capture fish as they swim into rotating baskets or compartments, allowing for continuous and passive monitoring of fish populations and migrations in rivers.'
+    ),
+    (
+        'Egg Mats',
+        'Egg mats are used in sample collection to provide a substrate for fish eggs to adhere to, enabling researchers to monitor and assess spawning activities and egg deposition in aquatic environments.'
+    ),
+    (
+        'Setline',
+        'Set lines are used in sample collection to capture fish by baiting hooks attached to a long fishing line anchored in place, allowing for targeted sampling of specific fish species over a period of time.'
```
Comment (mauberti-bc) [inline database/src/migrations/20250527114701_update_survey_methods.ts]: Should set lines be one word in the description, too?

File: database/src/migrations/20250527114701_update_survey_methods.ts (hunk 2 above)
Comment (LouisThedroux) [reply database/src/migrations/20250527114701_update_survey_methods.ts]: Ive had a go at this and repushed. All looks good to me

**Pull request closed without merging**